java-frontend/src/main/java/org/sonar/java/se/checks/CustomUnclosedResourcesCheck.java

```diff
@@ -119,7 +121,15 @@ private static String name(Tree tree) {
     if (tree.is(Tree.Kind.NEW_CLASS)) {
       return ((NewClassTree) tree).symbolType().name();
     }
-    return ((MethodInvocationTree) tree).symbolType().name();
+    MethodInvocationTree mit = (MethodInvocationTree) tree;
```

Comment:
The symbolType of the method is its return type, and therefore the reason why you are getting `void`. Why not using the type of the owner of the method in such cases? You will probably be more precise.

I would rewrite the method like this:
```
  MethodInvocationTree mit = (MethodInvocationTree) tree;
  if (mit.symbolType().isVoid()) {
      return mit.symbol().owner().type().name();
    }
    return mit.symbolType().name();
}
```